mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/ArenaAdapterService.kt

```diff
@@ -115,4 +125,99 @@ class ArenaAdapterService(
     fun removeDeltaker(id: UUID): QueryResult<Unit> {
         return query { deltakere.delete(id) }
     }
+
+    private fun lookForExistingAvtale(tiltaksgjennomforing: ArenaTiltaksgjennomforingDbo): UUID? {
+        val tiltakstype = tiltakstyper.get(tiltaksgjennomforing.tiltakstypeId)
+            ?: throw IllegalStateException("Ukjent tiltakstype id=${tiltaksgjennomforing.tiltakstypeId}")
+
+        return if (Tiltakskoder.isTiltakMedAvtalerFraMulighetsrommet(tiltakstype.arenaKode)) {
+            tiltaksgjennomforinger.get(tiltaksgjennomforing.id)?.avtaleId ?: tiltaksgjennomforing.avtaleId
+        } else {
+            tiltaksgjennomforing.avtaleId
+        }
+    }
+
+    private fun shouldBeManagedInSanity(gjennomforing: TiltaksgjennomforingAdminDto): Boolean {
+        val sluttDato = gjennomforing.sluttDato
+        return isEgenRegiTiltak(gjennomforing.tiltakstype.arenaKode) &&
+            (sluttDato == null || sluttDato.isAfter(TiltaksgjennomforingSluttDatoCutoffDate))
+    }
+
+    private fun maybeNotifyRelevantAdministrators(avtale: AvtaleAdminDto) {
+        val overordnetEnhet = resolveRelevantNavEnhet(
+            avtale.arenaAnsvarligEnhet,
+            avtale.administratorer,
+        ) ?: return
+
+        notifyRelevantAdministrators(overordnetEnhet) { administrators ->
+            ScheduledNotification(
+                type = NotificationType.TASK,
+                title = """Avtalen "${avtale.navn}" har endringer fra Arena, men mangler en ansvarlig administrator.""",
+                description = "Du har blitt varslet fordi din NAV-hovedenhet og avtalens ansvarlige NAV-enhet begge er relatert til ${overordnetEnhet.navn}.",
+                metadata = NotificationMetadata(
+                    linkText = "Gå til avtalen",
+                    link = "/avtaler/${avtale.id}",
+                ),
+                targets = administrators,
+                createdAt = Instant.now(),
+            )
+        }
+    }
+
+    private fun maybeNotifyRelevantAdministrators(gjennomforing: TiltaksgjennomforingAdminDto) {
+        val enhet = resolveRelevantNavEnhet(
+            gjennomforing.arenaAnsvarligEnhet,
+            gjennomforing.administratorer,
+        ) ?: return
+
+        notifyRelevantAdministrators(enhet) { administrators ->
+            ScheduledNotification(
+                type = NotificationType.TASK,
+                title = """Gjennomføringen "${gjennomforing.navn}" har endringer fra Arena, men mangler en ansvarlig administrator.""",
+                description = "Du har blitt varslet fordi din NAV-hovedenhet og gjennomføringens ansvarlige NAV-enhet begge er relatert til ${enhet.navn}.",
```

Comment:
Samme her som for avtale-teksten. Jeg savner noe om hva jeg som bruker bør gjøre.